### PR TITLE
Allure test reports

### PR DESCRIPTION
**Pull request by szczygiel-m**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -69,7 +69,8 @@ allprojects {
             junit_jupiter     : '5.10.2',
             testcontainers    : '1.19.8',
             spring            : '3.2.4',
-            assertj           : '3.25.3'
+            assertj           : '3.25.3',
+            allure            : '2.25.0'
```
Comment (WojciechSova) [inline build.gradle:73]: 2.29.0 is the lastest version

File: build.gradle [hunk 2]
```diff
@@ -87,6 +88,15 @@ allprojects {
...
         testImplementation group: 'org.awaitility', name: 'awaitility', version: '4.2.1'
 
+        // Allure Spock adapter
+        testImplementation(platform("io.qameta.allure:allure-bom:${versions.allure}"))
+        testImplementation("io.qameta.allure:allure-spock2")
+        testImplementation("io.qameta.allure:allure-junit-platform")
+
+        // Spock framework
+        testImplementation(platform("org.spockframework:spock-bom:${versions.spock}"))
+        testImplementation("org.spockframework:spock-core")
```
Comment (WojciechSova) [inline build.gradle:98]: Can you test if it will work without this dependencies?

**Review by WojciechSova — COMMENTED** [review]
Did you test the changes on UI on dev env?

**Review by WojciechSova — APPROVED** [review]

**Review by WojciechSova — APPROVED** [review]

File: integration-tests/build.gradle [hunk 3]
```diff
@@ -24,6 +31,13 @@ dependencies {
         exclude module: 'hornetq-native'
     }
 
+    // Import allure-bom to ensure correct versions of all the dependencies are used
+    testImplementation(platform("io.qameta.allure:allure-bom:$versions.allure"))
+    // Add necessary Allure dependencies to dependencies section
+    testImplementation("io.qameta.allure:allure-junit5")
+
+    agent "org.aspectj:aspectjweaver:1.9.21"
```
Comment (pitagoras3) [inline integration-tests/build.gradle:39]: **question:** Do we need to apply aspectJ agent? If we won't use it, do we loose something?

**Review by pitagoras3 — CHANGES_REQUESTED** [review]

File: integration-tests/build.gradle (hunk 3 above)
Comment (szczygiel-m) [reply integration-tests/build.gradle:39]: According to the [documentation](https://allurereport.org/docs/junit5/#configure-aspectj):
`Allure leverages AspectJ for the functionality of @Step and @Attachment annotations. Additionally, some framework integrations (such as [allure-assertj](https://mvnrepository.com/artifact/io.qameta.allure/allure-assertj)) rely on AspectJ integration to function correctly.`

**Review by moscicky — APPROVED** [review]

**Review by WojciechSova — APPROVED** [review]

**Review by pitagoras3 — APPROVED** [review]

**Pull request merged**